### PR TITLE
test(multiple): add performance testing page to dev-app

### PR DESCRIPTION
**Pull request by wagnermaciel**
* provides easy setup for debugging the rendering speed of our components

### CODE REVIEW

File: src/dev-app/dev-app/dev-app-layout.html [hunk 1]
```diff
@@ -1,31 +1,36 @@
...
   <mat-sidenav #navigation role="navigation">
     <mat-nav-list>
-      <a *ngFor="let navItem of navItems"
-         mat-list-item
-         (click)="navigation.close()"
-         routerLinkActive
-         #routerLinkActiveInstance="routerLinkActive"
-         [attr.tabindex]="routerLinkActiveInstance.isActive ? 0 : -1"
-         [routerLink]="[navItem.route]">
+      <a
```
Comment (crisbeto) [inline src/dev-app/dev-app/dev-app-layout.html:4]: The changes in this file all look unrelated.

File: src/dev-app/performance/performance-demo.html [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+>
+  Clear Metrics
+</button>
+
+<button color="primary" mat-raised-button (click)="runBenchmark()" [disabled]="isRunningBenchmark">
+  Run Benchmark
+</button>
+
+<ng-container *ngIf="show">
+  <ng-container *ngFor="let _ of [].constructor(componentCount)">
```
Comment (crisbeto) [inline src/dev-app/performance/performance-demo.html:74]: I think that this is going to cause a bunch of unnecessary change detections, because `[].constructor` will produce a new array any time it is checked.

File: src/dev-app/performance/performance-demo.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    this.dataSource.data = [];
+    this.totalRenderTime = this.getTotalRenderTime();
+  }
+
+  recordSample(): Promise<number> {
+    return new Promise(res => {
+      setTimeout(() => {
+        this.show = true;
+        const start = performance.now();
+        this._ngZone.onStable.pipe(take(1)).subscribe(() => {
```
Comment (crisbeto) [inline src/dev-app/performance/performance-demo.ts:128]: It's your call, but as I mentioned before over DM, I don't think that we should rely on zone.js to tell us when to measure, because it can be affected by more than our own code. If we benchmark through the dev tools, it'll be easier to discern what is slowing down a component's rendering.

File: src/dev-app/dev-app/dev-app-layout.html (hunk 1 above)
Comment (wagnermaciel) [reply src/dev-app/dev-app/dev-app-layout.html]: I think the changes from prettier made it hard to see. I've reverted the formatting to make it clear what has changed

File: src/dev-app/performance/performance-demo.ts (hunk 3 above)
Comment (mmalerba) [reply src/dev-app/performance/performance-demo.ts:132]: Maybe just add a disclaimer in he UI? e.g. "This is an approximate measure, for more precise performance benchmarking use Chrome devtools"

**Review by mmalerba — APPROVED** [review]
LGTM
